plugins/generic/driver/DRIVERDAO.php

```diff
@@ -50,7 +50,7 @@ public function &getDRIVERRecordsOrIdentifiers($setIds, $from, $until, $offset,
     {
         $records = [];
 
-        $result = new \ArrayIterator($this->_getRecordsRecordSetQuery($setIds, $from, $until, null)->get());
+        $result = new \ArrayIterator($this->_getRecordsRecordSetQuery($setIds, $from, $until, null)->get()->toArray());
```

Comment:
If you don't need the extra transformation, which is attempted at the `toArray()`, then the `all()` is a better fit :)

```suggestion
        $result = new \ArrayIterator($this->_getRecordsRecordSetQuery($setIds, $from, $until, null)->get()->all());
```